ansible_collections/arista/avd/plugins/action/eos_designs_structured_config.py

```diff
@@ -96,6 +97,8 @@ def run(self, tmp: Any = None, task_vars: dict | None = None) -> dict:
                 result=result,
                 templar=self.templar,
             )
+        except AristaAvdMissingVariableError as e:
+            raise AnsibleActionFail(message=str(e)) from e
         except Exception as error:
             raise AnsibleActionFail(message=str(error)) from error
```

Comment:
Should we not add the host field here too?

Reply:
it does not make sense for this part as we are already running it per host and so we are getting the proper error messages:


```
TASK [arista.avd.eos_designs : Generate device configuration in structured format] ***
ok: [legacy-autovpn-rr1 -> localhost]
ok: [legacy-autovpn-rr2 -> localhost]
ok: [legacy-autovpn-edge -> localhost]
ok: [legacy-autovpn-edge-no-default-policy -> localhost]
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: pyavd._errors.AristaAvdMissingVariableError: bgp_peer_groups.wan_overlay_peers.listen_range_prefixes
fatal: [cv-pathfinder-edge-no-default-policy -> localhost]: FAILED! => {"changed": false, "msg": "bgp_peer_groups.wan_overlay_peers.listen_range_prefixes"}
ok: [cv-pathfinder-edge-custom-default-policy -> localhost]
ok: [cv-pathfinder-edge2A -> localhost]
ok: [cv-pathfinder-edge2B -> localhost]
ok: [site-ha-enabled-leaf2A -> localhost]
ok: [site-ha-enabled-leaf2B -> localhost]
ok: [cv-pathfinder-transit1A -> localhost]
```